cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -351,5 +355,278 @@ ExtensionIdRegistry* default_extension_id_registry() {
   return &impl_;
 }
 
+Status FunctionMapping::AddArrowToSubstrait(std::string arrow_function_name, ArrowToSubstrait conversion_func){
+  if (arrow_to_substrait.find(arrow_function_name) != arrow_to_substrait.end()){
+    arrow_to_substrait[arrow_function_name] =  conversion_func;
+  }
```

Comment:
Should we return an invalid status as an else clause here?

Reply:
Made the change.

Reply:
Returning an AlreadyExist status.